function_calling/src/tools.py

```diff
@@ -152,40 +152,95 @@ class ReplSchema(BaseModel):
 ## SQL tool
 # tool schema
 class QueryDBSchema(BaseModel):
-    "A database querying tool. Use this to generate sql querys and retrieve the results from a database. Input should be a natural language question to the db."
+    "A query generation tool. Use this to generate sql querys and retrieve the results from a database. Do not pass sql querys directly Input must be a natural language question or instruction."
 
-    query: str = Field(..., description='python code to execute to evaluate')
+    query: str = Field(..., description='natural language question or instruction.')
+
+
+def sql_finder(text):
+    # print(text)
+    sql_code_pattern = re.compile(r'```sql\s+(.*?)\s+```', re.DOTALL)
```

Comment:
A small comment before these regexes (also below) to explain what we are looking for?

Reply:
added